app/src/usb.c

```diff
@@ -38,6 +38,22 @@ static const struct hid_ops ops = {
     .int_in_ready = in_ready_cb,
 };
 
+#ifdef CONFIG_ZMK_USB_REPORT_LEDS
+
+int zmk_usb_hid_receive_report(uint8_t *report, size_t len) {
+    k_sem_take(&hid_sem, K_MSEC(30));
```

Comment:
This looks copied from `zmk_usb_hid_send_report()`, but I'm not sure that code makes sense. 

If `hid_int_ep_read()` requires the semaphore to be taken, then I would expect us to check the return value of `k_sem_take()` to make sure we actually took the semaphore. As is, it looks like this code will try to take the semaphore for 30 ms, then continue regardless of whether it succeeded. Then if `hid_int_ep_read()` fails, it could give a semaphore it never took.

@petejohanson could you explain how the semaphore code in `zmk_usb_hid_send_report()` is supposed to work?